app/src/main/java/app/tivi/inject/NetworkModule.kt

```diff
@@ -16,21 +16,64 @@
 
 package app.tivi.inject
 
+import android.content.Context
 import app.tivi.BuildConfig
+import coil.util.CoilUtils
 import dagger.Module
 import dagger.Provides
+import java.util.concurrent.TimeUnit
 import javax.inject.Singleton
+import okhttp3.ConnectionPool
+import okhttp3.Dispatcher
+import okhttp3.OkHttpClient
 import okhttp3.logging.HttpLoggingInterceptor
+import okhttp3.logging.LoggingEventListener
 
 @Module
 class NetworkModule {
     @Singleton
     @Provides
-    fun provideHttpLoggingInterceptor(): HttpLoggingInterceptor {
+    fun provideHttpLoggingInterceptor(): HttpLoggingInterceptor? {
+        if (!BuildConfig.DEBUG) {
+            return null
+        }
+
         return HttpLoggingInterceptor().apply {
-            if (BuildConfig.DEBUG) {
-                level = HttpLoggingInterceptor.Level.HEADERS
-            }
+            level = HttpLoggingInterceptor.Level.HEADERS
+        }
+    }
+
+    @Singleton
+    @Provides
+    fun provideHttpEventListener(): LoggingEventListener.Factory? {
+        if (!BuildConfig.DEBUG) {
+            return null
         }
+
+        return LoggingEventListener.Factory()
+    }
+
+    @Singleton
+    @Provides
+    fun provideOkHttpClient(
+        httpLoggingInterceptor: HttpLoggingInterceptor?,
+        loggingEventListener: LoggingEventListener.Factory?,
+        applicationContext: Context
+    ): OkHttpClient {
+        return OkHttpClient.Builder()
+            .apply {
+                if (httpLoggingInterceptor != null) {
+                    addInterceptor(httpLoggingInterceptor)
+                }
+                if (loggingEventListener != null) {
+                    eventListenerFactory(loggingEventListener)
+                }
+            }
+            .cache(CoilUtils.createDefaultCache(applicationContext))
+            .connectionPool(ConnectionPool(10, 2, TimeUnit.MINUTES))
```

Comment:
Just wondering, where did these numbers come from?

Reply:
> Just wondering, where did these numbers come from?

Mainly the combination of two clients that default to 5 currently.  But actually it's 3 including Coil's OkHttp client.  And 2 minutes with an idle connection for the app seemed quite long, 5 (the default) seems wasteful for the radio and battery.

n.b. I'm mainly using the app to experiment with real world OkHttp performance, and how our logging and tools can be used.  So playing with some of them also.

Reply:
Sounds good to me!